src/mesh/mesh_base.C

```diff
@@ -505,8 +511,8 @@ void MeshBase::cache_elem_dims()
   // particular dimension have been deleted.
   _elem_dims.clear();
 
-  const_element_iterator el  = this->active_local_elements_begin();
-  const_element_iterator end = this->active_local_elements_end();
+  const_element_iterator el  = this->active_elements_begin();
+  const_element_iterator end = this->active_elements_end();
 
```

Comment:
We really ought to add an iterator which traverses both active and unpartitioned elements...